quay/quay
# Fix the RepoMirrorConfig model to match the latest migrations

**Issue:** https://issues.redhat.com/browse/PROJQUAY-940

**Changelog:** 

**Docs:** 

**Testing:** : 
- Create robot (user or org)
- Create mirror with previous robot
- Try to delete robot.

Expected result: API call should return 400 (Error msg pop up in UI)

**Details:** 
RepoMirrorConfig in the current database migration version has a
non-null constraint on the internal_robot field, but the model in
database.py does not.

Updates the  model to match the current database revision, and handles
delete api calls when there are mirrors still using the robot.

Also set a default test DATABASE_SECRET_KEY when generating the test.db
